testing/test_collection.py

```diff
@@ -809,6 +809,40 @@ def test___repr__():
         reprec = testdir.inline_run("-k repr")
         reprec.assertoutcome(passed=1, failed=0)
 
+    def test_keyword_matching_is_case_insensitive_by_default(self, testdir):
+        """Check that selection via -k EXPRESSION is case-insensitive.
+
+        Since markers are also added to the node keywords, they too can
+        be matched without having to think about case sensitivity.
+
+        """
+        testdir.makepyfile(
+            """
+            import pytest
+
+            def test_sPeCiFiCToPiC_1():
+                assert True
+
+            class TestSpecificTopic_2:
+                def test(self):
+                    assert True
+
+            @pytest.mark.sPeCiFiCToPic_3
+            def test():
+                assert True
+
+            @pytest.mark.sPeCiFiCToPic_4
+            class Test:
+                def test(self):
+                    assert True
+
```

Comment:
I suggest adding a failing test here for completeness:

```python
def test_failing():
    assert False
```

The outcome should still be the same, as that test should never be selected.

Reply:
:heavy_check_mark: